mlir/lib/Dialect/LLVMIR/Transforms/OpenMPOffloadPrivatizationPrepare.cpp

```diff
@@ -0,0 +1,423 @@
+            // which is stored at the var_ptr_ptr operand of %1. Then we need to
+            // copy this pointer to the new heap allocated memory location.
+            // Then, we need to store the address of the new heap location in
+            // the originating struct/descriptor. So, we generate the following
+            // (pseudo) MLIR code (Using the same names of mlir::Value instances
+            // in the example as in the code below)
+            //
+            // %dataMalloc = malloc(totalSize)
+            // %loadDataPtr = load %0 : !llvm.ptr -> !llvm.ptr
+            // memcpy(%dataMalloc, %loadDataPtr, totalSize)
+            // %newVarPtrPtrOp = llvm.getelementptr %heapMem[0, 0]
+            // llvm.store %dataMalloc, %newVarPtrPtrOp
+            // %1.cloned = omp.map.info var_ptr(%heapMem : !llvm.ptr, i32) ...
+            //                          var_ptr_ptr(%newVarPtrPtrOp : !llvm.ptr)
+            // %2.cloned = omp.map.info var_ptr(%heapMem : !llvm.ptr,
+            //                                             !desc_type)>) ...
+            //                          members(%1.cloned : [0] : !llvm.ptr)
+            //             -> !llvm.ptr
+            // omp.target nowait map_entries(%2.cloned -> %arg5,
+            //                               %1.cloned -> %arg8 : ..)
+            //            private(@privatizer %heapMem -> .. [map_idx=1] : ..) {
+
+            if (isFirstPrivate) {
+              assert(!memberMapInfoOp.getBounds().empty() &&
+                     "empty bounds on member map of firstprivate variable");
+              mlir::Location loc = memberMapInfoOp.getLoc();
+              mlir::Value totalSize =
+                  getSizeInBytes(memberMapInfoOp, mod, rewriter);
+              auto dataMalloc = allocateHeapMem(loc, totalSize, mod, rewriter);
+              auto loadDataPtr = rewriter.create<LLVM::LoadOp>(
+                  loc, memberMapInfoOp.getVarPtrPtr().getType(),
+                  memberMapInfoOp.getVarPtrPtr());
+              LLVM_ATTRIBUTE_UNUSED auto memcpy =
+                  rewriter.create<mlir::LLVM::MemcpyOp>(
+                      loc, dataMalloc.getResult(), loadDataPtr.getResult(),
+                      totalSize, /*isVolatile=*/false);
+              Operation *newVarPtrPtrOp = rewriter.clone(*varPtrPtrdefOp);
+              rewriter.replaceAllUsesExcept(memberMapInfoOp.getVarPtrPtr(),
+                                            newVarPtrPtrOp->getOpResult(0),
+                                            loadDataPtr);
+              rewriter.modifyOpInPlace(newVarPtrPtrOp, [&]() {
+                newVarPtrPtrOp->replaceUsesOfWith(varPtr, heapMem);
+              });
+              LLVM_ATTRIBUTE_UNUSED auto storePtr =
+                  rewriter.create<LLVM::StoreOp>(loc, dataMalloc.getResult(),
+                                                 newVarPtrPtrOp->getResult(0));
```

Comment:
```suggestion
              (void)rewriter.create<LLVM::StoreOp>(loc, dataMalloc.getResult(),
                                                 newVarPtrPtrOp->getResult(0));
```
to discard a `[[nodiscard]]` warning